src/net_processing.cpp

```diff
@@ -655,6 +655,18 @@ class PeerManagerImpl final : public PeerManager
     bool AlreadyHaveTx(const GenTxid& gtxid)
         EXCLUSIVE_LOCKS_REQUIRED(cs_main, !m_recent_confirmed_transactions_mutex);
 
+    /**
+     * Map of seen chain tip sets keyed by unique network ID.
+     *
+     * The size of each chain tip set is limited to the number of chain tips in
+     * our global block index.
+     */
+    std::map<uint64_t, std::set<uint256>> m_chain_tip_sets GUARDED_BY(cs_main);
```

Comment:
`cs_main` is already held in all the relevant locations so I thought I might as well use `cs_main` here instead of introducing another mutex (not even sure if that would be needed).